modules/vaos/spec/request/appointment_requests_request_spec.rb

```diff
@@ -13,78 +13,146 @@
-          .to eq('The required parameter "start_date", is missing')
+          .to eq('You do not have access to online scheduling')
       end
     end
+  end
 
-    context 'without an end_date' do
-      it 'has a parameter missing exception' do
-        get '/v0/vaos/appointment_requests', params: params.except(:end_date)
-        expect(response).to have_http_status(:bad_request)
-        expect(response.body).to be_a(String)
-        expect(JSON.parse(response.body)['errors'].first['detail'])
-          .to eq('The required parameter "end_date", is missing')
+  context 'loa3 user' do
+    let(:current_user) { build(:user, :mhv) }
+
+    describe 'GET appointment_requests' do
+      let(:start_date) { Date.parse('2019-08-20') }
+      let(:end_date) { Date.parse('2020-08-22') }
+      let(:params) { { start_date: start_date, end_date: end_date } }
+
+      context 'with flipper disabled' do
+        it 'does not have access' do
+          Flipper.disable('va_online_scheduling')
+          get '/v0/vaos/appointment_requests'
+          expect(response).to have_http_status(:forbidden)
+          expect(JSON.parse(response.body)['errors'].first['detail'])
+            .to eq('You do not have access to online scheduling')
+        end
       end
-    end
 
-    context 'with an invalid start_date' do
-      it 'has an invalid field type exception' do
-        get '/v0/vaos/appointment_requests', params: params.merge(start_date: 'invalid')
-        expect(response).to have_http_status(:bad_request)
-        expect(response.body).to be_a(String)
-        expect(JSON.parse(response.body)['errors'].first['detail'])
-          .to eq('"invalid" is not a valid value for "start_date"')
+      context 'without a start_date' do
+        it 'has a parameter missing exception' do
+          get '/v0/vaos/appointment_requests', params: params.except(:start_date)
+          expect(response).to have_http_status(:bad_request)
+          expect(response.body).to be_a(String)
+          expect(JSON.parse(response.body)['errors'].first['detail'])
+            .to eq('The required parameter "start_date", is missing')
+        end
       end
-    end
 
-    context 'with an invalid end_date' do
-      it 'has an invalid field type exception' do
-        get '/v0/vaos/appointment_requests', params: params.merge(end_date: 'invalid')
-        expect(response).to have_http_status(:bad_request)
-        expect(response.body).to be_a(String)
-        expect(JSON.parse(response.body)['errors'].first['detail'])
-          .to eq('"invalid" is not a valid value for "end_date"')
+      context 'without an end_date' do
+        it 'has a parameter missing exception' do
+          get '/v0/vaos/appointment_requests', params: params.except(:end_date)
+          expect(response).to have_http_status(:bad_request)
+          expect(response.body).to be_a(String)
+          expect(JSON.parse(response.body)['errors'].first['detail'])
```

Comment:
Could the invalid cases be wrapped in another description/context and have `JSON.parse(response.body)['errors'].first['detail']` be a let?